express/component_spec.py

```diff
@@ -0,0 +1,328 @@
+    """
+
+    name: str
+    description: str
+
+
+T = t.TypeVar("T", Input, Output)
+
+
+@dataclass
+class KFPComponent:
+    """
+    A class representing a Kubeflow Pipelines component.
+    Attributes:
+        name: The name of the component.
+        description: The description of the component.
+        image: The Docker image url for the component.
+        args: A dictionary of component arguments.
+        inputs: The input parameters for the component.
+        outputs: The output parameters for the component.
+        command: The command to run the component.
+    """
+
+    name: str
+    description: str
+    image: str
+    args: t.Dict[str, t.Dict[str, str]]
+    inputs: t.Union[Input, t.List[Input]] = None
+    outputs: t.Union[Output, t.List[Output]] = None
+    command: t.List[t.Union[str, t.Dict[str, str]]] = None
+
+    def __post_init__(self):
+        self.inputs = self._add_defaults_values(
+            self.inputs, self._get_default_inputs(), Input
+        )
+
+        self.outputs = self._add_defaults_values(
+            self.outputs, self._get_default_outputs(), Output
+        )
+
+        self._add_input_args()
+        self._set_component_run_cmd()
+
+    @staticmethod
+    def _get_default_inputs() -> t.List[Input]:
+        """
+        Set default inputs to components
+        """
+        inputs = [
+            Input(
+                name="input_manifest_path",
+                description="Path to the the input manifest",
+                type="String",
+            ),
+            Input(
+                name="args",
+                description="The extra arguments passed to the component",
+                type="String",
+            ),
+        ]
+        return inputs
+
+    @staticmethod
+    def _get_default_outputs() -> t.List[Output]:
+        """
+        Set default output to components
+        """
+        outputs = [
+            Output(
+                name="output_manifest_path",
+                description="The path to the output manifest",
+            )
+        ]
+        return outputs
+
+    @staticmethod
+    def _add_defaults_values(
+        values: t.Union[None, T, t.List[T]],
+        default_values: t.List[T],
+        value_type: t.Type[T],
+    ) -> t.List[T]:
+        """Add default values to a input/output list attribute"""
+
+        if values is None:
+            values = []
+
+        if isinstance(values, value_type):
+            values = [values]
+
+        values.extend(default_values)
+        return values
+
+    def _add_input_args(self):
+        """Add specified component arguments to the input"""
+        for arg_name, arg_info in self.args.items():
+            self.inputs.append(
+                Input(
+                    name=arg_name.strip(),
+                    description=arg_info["description"].strip(),
+                    type=get_kubeflow_type(arg_info["type"].strip()),
+                )
+            )
+
+    def _set_component_run_cmd(self):
```

Comment:
```suggestion
    def _set_component_run_command(self):
```
Let's use clear names everywhere